jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/core/autobahn/AutobahnWebSocketClient.java

```diff
@@ -132,121 +109,110 @@ public static void main(String[] args)
-                client.updateStatus("Will run all %d cases ...", caseCount);
+                LOG.info("Will run all {} cases ...", caseCount);
                 for (int caseNum = 1; caseNum <= caseCount; caseNum++)
                 {
-                    client.updateStatus("Running case %d (of %d) ...", caseNum, caseCount);
+                    LOG.info("Running case {} (of {}) ...", caseNum, caseCount);
                     client.runCaseByNumber(caseNum);
                 }
             }
             else
             {
-                client.updateStatus("Will run %d cases ...", caseNumbers.length);
+                LOG.info("Will run %d cases ...", caseNumbers.length);
                 for (int caseNum : caseNumbers)
                 {
                     client.runCaseByNumber(caseNum);
                 }
             }
-            client.updateStatus("All test cases executed.");
+            LOG.info("All test cases executed.");
             client.updateReports();
         }
         catch (Throwable t)
         {
-            t.printStackTrace(System.err);
+            LOG.warn("Test Failed", t);
         }
         finally
         {
             if (client != null)
-            {
                 client.shutdown();
-            }
         }
-        System.exit(0);
     }
 
-    private Logger log;
+    private static final Logger LOG = Log.getLogger(AutobahnWebSocketClient.class);
     private URI baseWebsocketUri;
     private WebSocketCoreClient client;
-    private String hostname;
-    private int port;
     private String userAgent;
 
     public AutobahnWebSocketClient(String hostname, int port, String userAgent) throws Exception
     {
-        this.log = Log.getLogger(this.getClass());
-        this.hostname = hostname;
-        this.port = port;
         this.userAgent = userAgent;
         this.baseWebsocketUri = new URI("ws://" + hostname + ":" + port);
         this.client = new WebSocketCoreClient();
-
-        // TODO: this should be enabled by default
-        // this.client.getExtensionFactory().register("permessage-deflate",PerMessageDeflateExtension.class);
         this.client.start();
     }
 
-    public int getCaseCount() throws IOException, InterruptedException, ExecutionException, TimeoutException
+    public int getCaseCount() throws IOException, InterruptedException
     {
         URI wsUri = baseWebsocketUri.resolve("/getCaseCount");
-        GetCaseCountHandler onCaseCount = new GetCaseCountHandler();
+        TestMessageHandler onCaseCount = new TestMessageHandler();
         Future<FrameHandler.CoreSession> response = client.connect(onCaseCount, wsUri);
 
         if (waitForUpgrade(wsUri, response))
         {
-            onCaseCount.awaitMessage();
-            if (onCaseCount.hasCaseCount())
-            {
-                return onCaseCount.getCaseCount();
-            }
+            String msg = onCaseCount.textMessages.poll(2, TimeUnit.SECONDS);
```

Comment:
2 seconds is a funny time to wait in a test.  If you expect it not to time out, then use something that is essentially infinity 10s or more, as 2s can happen in a bad GC or a busy CI machine working on something else.   If you expect it to timeout, then 2s is a long time to wait and will make total test time slow.